### PR TITLE
Do Not Override NPM Tag Based on Branch Name

### PR DESCRIPTION
**Pull request by NickGerleman**
Even the legacy CLI doesn't seem to rely on `v${branch}` tags (or current ore master), and they trample over anything we set in package.json. Stop tagging these so we an clean up npm tags.

While we're here, we try to inject the current branch name into more brachball logic in CI so we can avoid more changes while branching to stable releases.

Will cherry-pick this into 0.60-stable and 0.61-stable once in master.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4434)

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -205,7 +205,7 @@ jobs:
         run: yarn install --frozen-lockfile
 
       - name: Check for change files
-        run: node ./node_modules/beachball/bin/beachball.js check --changehint "Run `yarn change` from root of repo to generate a change file."
+        run: node ./node_modules/beachball/bin/beachball.js check --branch origin/$(Build.SourceBranchName) --changehint "Run `yarn change` from root of repo to generate a change file."
```
Comment (acoates-ms) [inline .github/workflows/pr.yml]: Build.SourceBranchName is 'merge' on PRs.  So I dont think this is doing the right thing..

Comment (acoates-ms) [reply .github/workflows/pr.yml]: There is System.PullRequest.TargetBranch which is similar but will be 'refs/heads/master' rather than just 'master'

Comment (NickGerleman) [reply .github/workflows/pr.yml]: Looks like you're right. I'll try to find if there's an env variable we can still use (Haven't tried to make things super correct for the GH version though, but we see this with the ADO one).

Comment (NickGerleman) [reply .github/workflows/pr.yml]: Turns out the docs lie, and the output of System.PullRequest.TargetBranch is just master. Makes things easier for us I guess.

Comment (NickGerleman) [reply .github/workflows/pr.yml]: Filed a bug at https://github.com/microsoft/azure-pipelines-tasks/issues/12588 on this being inconsistent with documentation. For now this seems to work though.

**Pull request merged**